src/test/java/de/tum/in/www1/artemis/LtiIntegrationTest.java

```diff
@@ -170,6 +173,23 @@ private void addJiraMocks(String email, String existingUser) throws Exception {
         jiraRequestMockProvider.mockAddUserToGroup("tumuser", false);
     }
 
+    private void addBitbucketMock(String requestBody) throws URISyntaxException {
+        bitbucketRequestMockProvider.enableMockingOfRequests();
+
+        String username = "prefix_";
+        Matcher matcher = Pattern.compile("lis_person_sourcedid=([^&#]+)").matcher(requestBody);
+        if (matcher.find() && !matcher.group(1).isEmpty()) {
+            username += matcher.group(1);
+        }
+        else {
+            matcher = Pattern.compile("ext_user_username=([^&#]+)").matcher(requestBody);
+            if (matcher.find()) {
```

Comment:
is it possible that matcher.find() returns false? and if thats the case what would happen to username?

Reply:
This method is a helper that takes certain input parameters. We know that in a standard request, the username field is send with lis_person_sourcedid for edX, for moodle with ext_user_username. In this scenario it is not the case that match.find() would be false.